webrender/src/resource_cache.rs

```diff
@@ -294,7 +297,17 @@ impl ResourceCache {
     }
 
     pub fn delete_image_template(&mut self, image_key: ImageKey) {
-        self.image_templates.remove(&image_key);
+        let value = self.image_templates.remove(&image_key);
+
+        // If the key is associate to an external image, pass the external id to render for cleanup.
+        if let Some(image) = value {
```

Comment:
Let's print a error if image is not Some(..) here as well, since that indicates a bug (such as double deletion of an image key).